app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,25 @@
+class MessagesController < ApplicationController
+  before_action :authenticate_user!
+  respond_to :json
+
+  def create
+    @message = Message.new(message_params)
```

Comment:
Maybe there is a reason but why you just do something like this:
```
message = Message.create!(message_params)
broadcast_message(message)
head :ok
````
Notice the `!` in the create method, if it fails, it will raise an exception and will be catched by the api_controller and then render a bad_request status

Reply:
Update: i just realized that as this project is not made with the api_base, you dont have an ApiController managing the exceptions. You can leave it as it's made right now but keep in mind what I said as an improvement for the future (I know this project is just for learning and won't have any upgrade in the future haha)